### PR TITLE
Game over screen added

### LINKED ISSUE #29: Game over screen

### PR DESCRIPTION
**Pull request by guilhermehto**
When the whole party is defeated, a game over screen is displayed. The player can either restart the encounter, or exit the game from it.

#29

### CODE REVIEW

File: godot/Game.gd [hunk 1]
```diff
@@ -10,6 +10,7 @@ onready var party = $Party as Party
 onready var music_player = $MusicPlayer
 
 var transitioning = false
+var ongoing_combat
```
Comment (NathanLovato) [inline godot/Game.gd]: Should be `combat_arena`: that's the name of the node and scene type that this variable holds

File: godot/Game.gd [hunk 2]
```diff
@@ -27,28 +28,33 @@ func enter_battle(formation: Formation):
...
 	yield(transition.fade_to_color(), "completed")
 	remove_child(local_map)
-	var combat_arena = combat_arena_scene.instance()
-	add_child(combat_arena)
-	combat_arena.connect("victory", self, "_on_CombatArena_player_victory")
-	combat_arena.initialize(formation, party.get_active_members())
+	ongoing_combat = combat_arena_scene.instance()
+	add_child(ongoing_combat)
+	ongoing_combat.connect("victory", self, "_on_CombatArena_player_victory")
+	ongoing_combat.connect("combat_restarted", self, "_on_combat_restarted")
```
Comment (NathanLovato) [inline godot/Game.gd]: the signal should be like `restart_requested`: triggering this signal causes the encounter to restart

File: godot/combat/CombatArena.gd [hunk 3]
```diff
@@ -5,17 +5,22 @@ const BattlerNode = preload("res://combat/battlers/Battler.tscn")
 onready var turn_queue : TurnQueue = $TurnQueue
 onready var interface = $CombatInterface
 onready var rewards = $Rewards
+onready var game_over_interface = $GameOverInterface as GameOverInterface
```
Comment (NathanLovato) [inline godot/combat/CombatArena.gd]: The game over UI shouldn't only be available during battle. It should be part of the main game scene. And so it shouldn't be controlled by the CombatArena either

Comment (NathanLovato) [reply godot/combat/CombatArena.gd]: There might be multiple reasons to trigger a game over other than losing an encounter: missing a timer during a mission, a random event...

File: godot/combat/CombatArena.gd [hunk 4]
```diff
@@ -80,13 +85,21 @@ func battle_end():
 	var active_battler = get_active_battler()
 	active_battler.selected = false
 	var player_won = active_battler.party_member
+	print("Valid: " + str(is_instance_valid(active_battler.party_member)))
```
Comment (NathanLovato) [inline godot/combat/CombatArena.gd]: Please remove your prints from final PRs

File: godot/combat/CombatArena.gd [hunk 5]
```diff
@@ -80,13 +85,21 @@ func battle_end():
...
 	if player_won:
 		emit_signal("victory")
 		yield(rewards.on_battle_completed(), "completed")
 		emit_signal("battle_ended", self.party)
 	else:
-		emit_signal("battle_ended", self.party)
-		emit_signal("gameover")
```
Comment (NathanLovato) [inline godot/combat/CombatArena.gd]: To me this PR shouldn't touch the combat arena, or affect it as little as possible. When the player loses, it's game over, the battle ends. Game opens the game over UI, from which the player can load a game.

**Comment by guilhermehto** [comment]
Here's how the menu looks now: 

694e3dd077073d45051c655f5229f1d2

I added an arrow to navigate through the options. There's touch, mouse and keyboard support. Any suggestions/feedback?

The displayed message is based on the reason why the game over is being shown, this way we can easily add different messages to the player.

The Game scene is now the one responsible to display the interface.

**Comment by NathanLovato** [comment]
Way better, thanks!

**Pull request merged**